amimof/multikube
# Order of Middlewares

**- What I did**
Kubernetes has many endpoints and APIs that not necessarily starts with /api. The `WithCtxRoot` middleware extracts clustername from the URL path by checking if it contains /api. This is not ideal since the Kubernetes apiserver could potentially expose any path which we do not want to care about.

**- How I did it**
* Re-ordered Middlewares. `WithCtxRoot` must come before `WithHeader`.
* Removed checking for /api string in URL path

**- How to verify it**
1. Make a call to Multikube with URL `/<context>/api/v1/namespaces` - apiserver should respond
2. Make a cal to Multikube with Http header `Multikube-Context: <context>` - apiserver should respond


**- Description for the CHANGELOG**
Improved URL path and HTTP header context routing
